Automattic/simplenote-macos
# Magic Links: Implement Code view and Password view

### Fix
Rounding third base on this one, so what we have here is I have finished splitting up the different auth views so that on each view you have one option.  So all the changes listed out here:

- Added auth with code view
- added auth with password view
- dropped access to password view in earlier views
- fixed issue with the button titles during animation
- added persistent state from one view to the next
- added authentication with a code
- Added header text to the different views

### Test
1. Launch simplenote and tap on Sign in.  Enter an invalid email and tap sign up.  Confirm that you see an error saying the email is invalid
2. Go back, tap on login. On the login with email screen, confirm that you can't use an invalid email without getting an error.  Attempt to enter a valid email address.  Confirm there is an animation and that you are pushed to the code view
3. Tap on the login with password button.  On the password view, confirm an invalid password shows an error and a successful password logs you in
4. Back on the code view confirm an invalid code fails to login and a valid one authenticates (note that we don't have handling for invalid codes yet or error handling for the code auth)

### Review
***(Required)*** Add instructions for reviewers.  For example:
> Only one developer is required to review these changes, but anyone can perform the review.

### Release
***(Required)*** Add a concise statement to `RELEASE-NOTES.txt` if the changes should be included in release notes. Include details about updating the notes in this section. For example:
> These changes do not require release notes.
